addons/account/models/account_move.py

```diff
@@ -4255,24 +4251,19 @@ def _disable_recursion(self, container, key, default=None, target=True):
         :return: True iff we should just exit the context manager
         """
 
-        disabled = container['records'].env.context.get(key, default) == target
-        previous_values = {}
-        previous_envs = set(self.env.transaction.envs)
-        if not disabled:  # it wasn't disabled yet, disable it now
-            for env in self.env.transaction.envs:
-                previous_values[env] = env.context.get(key, EMPTY)
-                env.context = frozendict({**env.context, key: target})
+        stack = self.env.cr.cache.setdefault('account_disable_recursion_stack', StackMap())
+        try:
+            current_val = stack[key]
+        except KeyError:
+            current_val = self.env.context.get(key, default)
+
+        disabled = current_val == target
+        top = stack.pushmap({key: target})
```

Comment:
From what I can see, `top` and `stack_top` aren't necessary.  By construction, both calls to `pushmap()` and `popmap()` must correspond to each other, since the stack is only manipulated from here.  So please remove your changes in the implementation of `StackMap`, as it is not necessary.  It only introduces a risk for a bug.

Reply:
I pushed it but I don't agree with it.
It's a nice check at no cost, which could be a hair-puller if something goes wrong because it isn't there.